File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text : ")

# letter, word, sentence 각각 count해야.(2020.09.24 @투썸플레이스 광화문)

letter = word = sentence = 0

for i in range(len(text)):
    if (text[i] >= 'a' and text[i] <= 'z') or (text[i] >= 'A' and text[i] <= 'Z'):
        letter += 1
    elif text[i] == ' ':
        word += 1
    elif text[i] == '.' or text[i] == '?' or text[i] == '!':
        sentence += 1
word += 1


#print (f"letter : {letter}, word : {word}, sentence : {sentence}\n")

L = (letter / word) * 100
S = (sentence / word) * 100

grade = (0.0588 * L - 0.296 * S - 15.8)
print(f"grade : {grade}\n")
if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print(f"Grade : {round(grade)}")


#L : the average number of letters per 100 words in the text
#S : the average number of sentences per 100 words in the text

File: cs50 IDE/readability/readability.c
//The Coleman-Liau index
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("text: ");
    int letters = 0;  // number of letters
    int words = 0;  // number of words
    int sentences = 0;  // number of sentences

    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] >= 65 && text[i] <= 90)
        {
            letters++;
        }
        else if (text[i] >= 97 && text[i] <= 122)
        {
            letters++;
        }
        else
        {
            if (text[i] == 32)
            {
                words++;
            }
            else if (text[i] == 33 || text[i] == 46 || text[i] == 63)
            {
                sentences++;
            }
        }
    }
    words = words + 1;

    //printf("Letters : %i\n",letters);
    //printf("Words : %i\n",words);
    //printf("Sentences : %i\n",sentences);

    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;

    //printf("L : %f\n", L);
    //printf("S : %f\n", S);

    int Grade = round(0.0588 * L - 0.296 * S - 15.8);
    if (Grade >= 16)
    {
        printf("Grade 16+");
    }
    else if (Grade < 1)
    {
        printf("Before Grade 1");
    }
    else
    {
        printf("Grade %i", Grade);
    }
    printf("\n");
}